website/addons/mendeley/views.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+from flask import request
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.project.decorators import must_be_contributor_or_public
```

Comment:
Why not

```python
from website.project.decorators import (
    must_be_contributor_or_public,
    must_have_permission,
    must_not_be_registration,
    must_have_addon,
)
```

This is more concise and consistent with much of the codebase (though I'm aware it's not completely consistent).